internal/transport/http2_client.go

```diff
@@ -983,6 +1027,18 @@ func (t *http2Client) Close(err error) {
 	}
 	t.mu.Unlock()
 	t.controlBuf.finish()
+	t.controlBuf.setError(nil)
+	err1 := t.controlBuf.put(&goAway{code: http2.ErrCodeNo, debugData: []byte("GOAWAY from client"), headsUp: true})
```

Comment:
After calling `controlBuf.finish()`, a `put` would just be a no-op. I think there are a couple of options here. 

One other way I can think about is, if we can make sure the loopywriter,run() has returned, it should be safe to access the framer directly and write the goAway frame directly. 

Or the other option is to somehow make sure we `put()` the GoAway and also assure that this is the last thing that is processed in the controlbuf.

Reply:
setError(nil) is only required since finish() sets the error which causes the put to be a no-op. I would prefer taking another approach.

Reply:
Sure, I have updated the PR